components/org.wso2.carbon.identity.conditional.auth.functions.acr/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/acr/SelectAcrFromFunction.java

```diff
@@ -0,0 +1,84 @@
+
+    public String evaluate(JsAuthenticationContext context,Object... possibleOutcomesObj) {
+
+        String[] possibleOutcomes = extractPossibleOutcomes(context, possibleOutcomesObj);
+        List<String> acrListRequested = context.getWrapped().getRequestedAcr();
+        if (acrListRequested == null || acrListRequested.isEmpty()) {
+            if (log.isDebugEnabled()) {
+                log.debug("ACR values from context is empty. Selecting the default outcome as null.");
+            }
+            return null;
+        }
+        if (possibleOutcomes.length > 0) {
+            return selectBestOutcome(acrListRequested, possibleOutcomes);
+        }
+        return null;
+    }
+
+    private String[] extractPossibleOutcomes(JsAuthenticationContext context, Object possibleOutcomesObj) {
+
+        String[] possibleOutcomes;
+        if (possibleOutcomesObj instanceof Object[]) {
+            Object[] possibleOutcomesObjArray = (Object[]) possibleOutcomesObj;
+            possibleOutcomes = Arrays.copyOf(possibleOutcomesObjArray, possibleOutcomesObjArray.length, String[].class);
+        } else if (possibleOutcomesObj instanceof ScriptObjectMirror) {
+            if (((ScriptObjectMirror) possibleOutcomesObj).isArray()) {
+                possibleOutcomes = ((ScriptObjectMirror) possibleOutcomesObj).to(String[].class);
+            } else {
+                log.error("Invalid argument provided for possible outcomes for " + AcrFunctionsServiceComponent
+                        .JS_FUNC_SELECT_ACR_FROM + " function in service provider: " + context.getWrapped()
+                        .getServiceProviderName() + ". Expected array of strings.");
+                possibleOutcomes = new String[0];
+            }
+        } else {
+            log.error("Invalid argument provided for possible outcomes for " + AcrFunctionsServiceComponent
+                    .JS_FUNC_SELECT_ACR_FROM + " function in service provider: " + context.getWrapped()
+                    .getServiceProviderName() + ". Expected array of strings.");
+            possibleOutcomes = new String[0];
+        }
+        return possibleOutcomes;
+    }
+
+    private String selectBestOutcome(List<String> acrListRequested, String[] possibleOutcomes) {
+
+        Map<Integer, String> acrRequestedWithPriority = new TreeMap<>(Collections.reverseOrder(
+                (Comparator<Integer>) (o1, o2) -> o2.compareTo(o1)));
+        String acrSelected = null;
+
+        for (String acrChecked : acrListRequested) {
+            for (int x = 0; x < possibleOutcomes.length; x++) {
+                String outcomeToTest = possibleOutcomes[x];
+                if (outcomeToTest.equals(acrChecked)) {
+                    if (log.isDebugEnabled()) {
+                        log.debug("Reassigning Best Match for the outcome : " + outcomeToTest + " with priority : " +
+                                x+1);
```

Comment:
possible format issue

Reply:
That was to keep the number of line characters less 120